samza-core/src/main/java/org/apache/samza/clustermanager/ClusterResourceManager.java

```diff
@@ -132,6 +132,9 @@ public ClusterResourceManager(Callback callback) {
 
   public abstract void stop(SamzaApplicationState.SamzaAppStatus status);
 
+  public boolean isResourceExpired(SamzaResource resource) {
```

Comment:
javadoc please
also, this method likely belongs in SamzaResource?
(if it really needs to be here, needs to be abstract)

Reply:
Kubernetes does not have a concept of expired allocated resource so this method needs a default implementation, it cannot belong to SamzaResource because that is yarn agnostic class and timeout is read from Yarn config and not a samza config.